tests/addhostCliTest.php

```diff
@@ -0,0 +1,201 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link http://librenms.org
+ * @copyright  2020 Lars Elgtvedt Susaas
+ * @author     Lars Elgtvedt Susaas
+ */
+
+/*
+    Usage (SNMPv1/2c)    : ./addhost.php [-g <poller group>] [-f] [-b] [-p <port assoc mode>] <hostname> [community] [v1|v2c] [port] [udp|udp6|tcp|tcp6]
+    Usage (SNMPv3)       :
+        Config Defaults  : ./addhost.php [-g <poller group>] [-f] [-b] [-p <port assoc mode>] <hostname> any v3 [user] [port] [udp|udp6|tcp|tcp6]
+        No Auth, No Priv : ./addhost.php [-g <poller group>] [-f] [-b] [-p <port assoc mode>] <hostname> nanp v3 [user] [port] [udp|udp6|tcp|tcp6]
+        Auth, No Priv    : ./addhost.php [-g <poller group>] [-f] [-b] [-p <port assoc mode>] <hostname> anp v3 <user> <password> [md5|sha] [port] [udp|udp6|tcp|tcp6]
+        Auth,    Priv    : ./addhost.php [-g <poller group>] [-f] [-b] [-p <port assoc mode>] <hostname> ap v3 <user> <password> <enckey> [md5|sha] [aes|des] [port] [udp|udp6|tcp|tcp6]
+    Usage (ICMP only)    : ./addhost.php [-g <poller group>] [-f] -P <hostname> [os] [hardware]
+
+    -g <poller group> allows you to add a device to be pinned to a specific poller when using distributed polling. X can be any number associated with a poller group
+    -f forces the device to be added by skipping the icmp and snmp check against the host.
+    -p <port assoc mode> allow you to set a port association mode for this device. By default ports are associated by 'ifIndex'.
+        For Linux/Unix based devices 'ifName' or 'ifDescr' might be useful for a stable iface mapping.
+        The default for this installation is 'ifIndex'
+        Valid port assoc modes are: ifIndex, ifName, ifDescr, ifAlias
+    -b Add the host with SNMP if it replies to it, otherwise only ICMP.
+    -P Add the host with only ICMP, no SNMP or OS discovery.
+*/
+
+namespace LibreNMS\Tests;
+
+use App\Models\Device;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use LibreNMS\Tests\DBTestCase;
+
+class AddhostCliTest extends DBTestCase
+{
+    use DatabaseTransactions;
+    private $hostName = "testHost";
+
+    public function tearDown() : void
+    {
+        exec("./delhost.php ".$this->hostName." ");
+        parent::tearDown();
+    }
+
+    public function testCLIsnmpV1()
+    {
+        exec("./addhost.php -g poller_group -f -p ifIndex ".$this->hostName." community v1 111 tcp");
```

Comment:
This should be `\Artisan::call('device:add', ...)`

Reply:
Sorry for late reply.
I will do the changes